src/cpu-template-helper/build.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#[cfg(target_arch = "aarch64")]
+use std::io::Write;
+
+const MOCK_KERNEL_PATH: &str = "src/utils/mock_kernel/kernel.bin";
+
+// Kernel header for aarch64 that comes from the kernel doc Documentation/arm64/booting.txt.
+#[cfg(target_arch = "aarch64")]
+#[derive(Default)]
+#[repr(C, packed)]
+struct KernelHeader {
+    code0: u32,       // Executable code
+    code1: u32,       // Executable code
+    text_offset: u64, // Image load offset,
+    image_size: u64,  // Effective Image size, little endian
+    flags: u64,       // kernel flags, little endian
+    res2: u64,        // reserved
+    res3: u64,        // reserved
+    res4: u64,        // reserved
+    magic: u32,       // Magic number, little endian, "ARM\x64"
+    res5: u32,        // reserved (used for PE COFF offset)
+}
+
+fn main() {
+    if cfg!(target_arch = "x86_64") {
+        println!("cargo:rerun-if-changed=src/utils/mock_kernel/main.c");
+        let status = std::process::Command::new("gcc")
+            .args([
+                // Do not use the standard system startup files or libraries when linking.
+                "-nostdlib",
+                // Prevents linking with the shared libraries.
+                "-static",
+                // Do not generate unwind tables.
+                "-fno-asynchronous-unwind-tables",
+                // Remove all symbol table and relocation information.
+                "-s",
+                "-o",
+                MOCK_KERNEL_PATH,
+                "src/utils/mock_kernel/main.c",
+            ])
+            .status()
+            .expect("Failed to execute gcc command");
+        if !status.success() {
+            panic!("Failed to compile mock kernel");
+        }
+    } else if cfg!(targert_arch = "aarch64") {
+        let header = KernelHeader {
+            magic: 0x644D5241,
+            ..std::default::Default::default()
+        };
+        // SAFETY: This is safe as long as `header` is valid as `KernelHeader`.
+        let header_bytes = unsafe {
+            std::slice::from_raw_parts(
+                (&header as *const KernelHeader).cast::<u8>(),
+                std::mem::size_of::<KernelHeader>(),
+            )
+        };
+
+        let mut file = std::fs::File::create(MOCK_KERNEL_PATH).expect("Failed to create a file");
+        file.write_all(header_bytes)
+            .expect("Failed to write kernel header to a file");
+    }
```

Comment:
Nit: This is a little paranoid but you could add:
```suggestion
    else {
        panic!("Unsupported arch");
    }
```

Reply:
The release script failed with the following error. Do you have any insight?
```
+ cargo build --target aarch64-unknown-linux-musl --release --workspace
warning: patch for `kvm-bindings` uses the features mechanism. default-features and features will not take effect because the patch dependency does not support this mechanism
   Compiling cpu-template-helper v1.8.0-dev (/firecracker/src/cpu-template-helper)
   Compiling serde v1.0.197
   Compiling thiserror v1.0.58
   Compiling bindgen v0.69.4
   Compiling userfaultfd-sys v0.5.0
   Compiling clap v4.5.3
   Compiling clippy-tracing v0.1.0 (/firecracker/src/clippy-tracing)
   Compiling vm-memory v0.14.1
   Compiling vm-allocator v0.1.1
error: failed to run custom build command for `cpu-template-helper v1.8.0-dev (/firecracker/src/cpu-template-helper)`

Caused by:
  process didn't exit successfully: `/firecracker/build/cargo_target/release/build/cpu-template-helper-10d08fee895bfd8f/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at src/cpu-template-helper/build.rs:63:9:
  Unsupported arch
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
wa
```

Reply:
How does rust pick "target_arch"?

Reply:
sorry, ignore the above. I just mistyped.